numba/cpython/randomimpl.py

```diff
@@ -1898,6 +1898,10 @@ def _impl(a):
                 U = 1.0 - np.random.random()
                 V = np.random.random()
                 X = int(math.floor(U ** (-1.0 / am1)))
+                
+                if (X > INT64_MAX or X < 1.0):
```

Comment:
I think this should be `LONG_MAX` not `INT64_MAX` as windows has a 32bit long?

Reply:
Fixed this in 
fca8cd8

Reply:
It's "system" dependent, perhaps use something like
```python
(1 << ( 8 * ctypes.sizeof(ctypes.c_long) - 1)) - 1
```
(there's a shorter way to do this in C but I'm not sure how well it translates to Python/`ctypes`).

Reply:
changed confirmed